src/cart/README.env

```diff
@@ -26,6 +26,13 @@ This file lists the environment variables used in CaRT.
    in D_LOG_FILE. If not defined or setting to zero value will result in this
    feature being disabled.
 
+ . D_LOG_FILE_APPEND_RANK
+   Set this to a non-zero value to create separate log files based upon the
+   process cart rank. When enabled this feature will rename the existing log file
+   and append rank to the filename upon the first invocation of rank setting.
+   Subsequent settings of the rank within the process are treated as noops.
+   As ranks are only set on the server, this is a server-only feature.
```

Comment:
If only the server is setting this then wouldn't it be better to export this as a c function rather than an environment variable?

Reply:
> If only the server is setting this then wouldn't it be better to export this as a c function rather than an environment variable?
not sure how a c function would help here since it will need to be conditionally called anyways based on either a yaml setting or an env, as not every test might want to enable this log renaming at runtime. 
plus this seemed consistent with all other log-related options that we provide (e.g. D_LOG_FILE_APPEND_PID)

Reply:
Is this for testing then?  The comment in the code implies that it's something the server would want to use.

What's going to set this env, and why?

Reply:
> Is this for testing then? The comment in the code implies that it's something the server would want to use.
> 
> What's going to set this env, and why?

This is to help with debugging on clusters that have lots of servers. We typically get logs in a form of <group_name>.log.<pid>, but then someone has to go through each log and figure out which rank what log corresponds to.  Instead when setting this envariable one can get rank automatically attached to the filename once the rank is determined.

I've included the example of how log files look like when running with this set.

Reply:
So why make this configurable via an env?  Shouldn't the server just set the log file automatically when it knows the pid?

Reply:
Some people might not want this and instead would want the log file to be deterministic and be exactly how they specified it in D_LOG_FILE. 
This is no different than any other option we expose, such as whether to attach PID to the log or not (controlled via D_LOG_FILE_ATTACH_PID env), or what to include in logs themselves.

Reply:
Ideally we would just support special characters as part of D_LOG_FILE e.g. D_LOG_FILE=~/my_log.%h.%r     or some such to expand into, for example, %h = hostname, %r = rank, etc... but this would be much bigger change.

Reply:
I can see an env makes sense for cart in isolation but from a daos perspective what we're saying is that there's a server option that has to be set in a config file, and if that option is set then the daos engine will communicate with itself, via an environment variable, that it wasn't this feature enabled.